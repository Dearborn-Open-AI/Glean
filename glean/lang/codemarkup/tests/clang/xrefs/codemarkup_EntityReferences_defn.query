query: codemarkup.EntityReferences { target = { cxx = { defn = _ } } }
perf: true
transform: [gensort, []]
