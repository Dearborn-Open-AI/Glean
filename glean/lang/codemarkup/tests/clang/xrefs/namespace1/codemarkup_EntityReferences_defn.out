[
  "@generated",
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "bar" },
                        "parent": { "key": { "name": { "key": "foo" } } }
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 18,
                      "columnBegin": 1,
                      "lineEnd": 23,
                      "columnEnd": 1
                    }
                  }
                },
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 366, "length": 3 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "bar" },
                        "parent": { "key": { "name": { "key": "foo" } } }
                      }
                    },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 18,
                      "columnBegin": 1,
                      "lineEnd": 23,
                      "columnEnd": 1
                    }
                  }
                },
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 395, "length": 3 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "foo" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 24,
                      "columnEnd": 1
                    }
                  }
                },
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 349, "length": 3 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "foo" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 24,
                      "columnEnd": 1
                    }
                  }
                },
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 361, "length": 3 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "foo" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 24,
                      "columnEnd": 1
                    }
                  }
                },
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 378, "length": 3 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "namespace_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": { "key": { "name": { "key": "foo" } } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 9,
                      "columnBegin": 1,
                      "lineEnd": 24,
                      "columnEnd": 1
                    }
                  }
                },
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 390, "length": 3 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "S" },
                        "scope": {
                          "namespace_": { "key": { "name": { "key": "foo" } } }
                        }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 11,
                      "columnBegin": 1,
                      "lineEnd": 11,
                      "columnEnd": 11
                    }
                  }
                },
                "bases": [ ],
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 354, "length": 1 } }
    }
  },
  {
    "key": {
      "target": {
        "cxx": {
          "defn": {
            "record_": {
              "key": {
                "declaration": {
                  "key": {
                    "name": {
                      "key": {
                        "name": { "key": "T" },
                        "scope": {
                          "namespace_": {
                            "key": {
                              "name": { "key": "bar" },
                              "parent": { "key": { "name": { "key": "foo" } } }
                            }
                          }
                        }
                      }
                    },
                    "kind": { "struct_": { } },
                    "source": {
                      "file": { "key": "test.cpp" },
                      "lineBegin": 20,
                      "columnBegin": 1,
                      "lineEnd": 20,
                      "columnEnd": 11
                    }
                  }
                },
                "bases": [ ],
                "members": { "key": [ ] }
              }
            }
          }
        }
      },
      "file": { "key": "test.cpp" },
      "range": { "span": { "start": 371, "length": 1 } }
    }
  }
]